pkg/backend/filestate/backend.go

```diff
@@ -487,6 +498,13 @@ func (b *localBackend) apply(
 	return changes, nil
 }
 
+// query executes a query program against the resource outputs of a locally hosted stack.
+func (b *localBackend) query(ctx context.Context, stack backend.Stack, op backend.UpdateOperation,
+	events chan<- engine.Event) result.Result {
+
+	panic("not implemented")
```

Comment:
oh right, I forgot about this, sorry. the PR is big enough that it's starting to get away from me.

Reply:
As for the second complaint, it's hard to know what you mean specifically here, but the somewhat awkward structure of the code is because the things we need are scattered between the engine and backend packages. We could perform more surgery and pull it out, but I personally don't think it's worth the trouble.

Reply:
Let's resolve for now, re-open if you have more you want to discuss.